dxp/aui_script.md

```diff
@@ -0,0 +1,148 @@
+
+Like this example:
+
+```jsp
+<aui:script>
+	Liferay.fire('myGlobalEvent', {
+		payload: 'myPayload',
+	});
+</aui:script>
+```
+
+it will be turned on:
+
+```js
+Liferay.fire('myGlobalEvent', {
+	payload: 'myPayload'
+});
+```
+
+#### Import an AUI/YUI module using the `use` property
+
+> ⚠️ This is a deprecated use case, though it still works if someone configures it.⚠️
+
+Each AUI / YUI module listed as a value for the `use` property will be [loaded asyncronously by the `AUI().use` function](https://github.com/yui/yui3/blob/master/src/yui/js/yui.js#L1323..L1331).
+
+You can import several YUI / AUI modules within the same `use` property, as in this example:
+
+Here is an example:
+
+```jsp
+<aui:script use="liferay-alert">
+	new Liferay.Alert(
+		{
+			closeable: true,
+			message: 'I like drink tea while eating french potatoes',
+			type: 'success'
+		}
+	).render('#myId');
+</aui:script>
+```
+
+The output will be a code like this:
+
+```js
+AUI().use('liferay-alert', function (A) {
+	(function() {var $ = AUI.$;var _ = AUI._;
+		new Liferay.Alert(
+			{
+				closeable: true,
+				message: 'I like drink tea while eating french potatoes',
+				type: 'success'
+			}
+		).render('#myId');
+	})()
+}
+```
+
+Note in the AlloyUI components, coming from the alloy-ui repository, classes are made available from the global variable `A` on the scope of the `aui:script`. Here as in this [code snippet the Menu component is defined](https://github.com/liferay/alloy-ui/blob/master/src/aui-menu/js/aui-menu.js#L39]).
+
+However, components made available from the `frontend-js-aui-web` module will be under the global Liferay variable. Like this [example here, `Liferay.Upload`](https://github.com/liferay/liferay-portal/blob/815f48f484351e18b61e4b9c9fbf40f0609bdc56/modules/apps/frontend-js/frontend-js-aui-web/src/main/resources/META-INF/resources/liferay/upload.js#L1541), used in the code example above.
+
+#### Import a ES module using `require` property
+
+Allows users to require a modern ES module as in the following example:
+
+Considering We already have `metal-clipboard` and `metal-dom` dependencies listed [on our module package.json](https://help.liferay.com/hc/en-us/articles/360018159771-The-Structure-of-OSGi-Bundles-Containing-npm-Packages-)
+
+By default the name of the loaded resource will be the transformed path to camel-case default exported module/file:
+
+```jsp
+<aui:script require="metal-clipboard/src/Clipboard">
+    // Note We have the provided path to require metal-clipboard/src/Clipboard normalized and camel-case as a variable name
```

Comment:
-> // Note how the required `metal-clipboard/src/Clipboard` is made available in the `metalClipboardSrcClipboard` variable

Reply:
✅